src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -862,6 +865,17 @@ void RNN::prepareParams() {
 
     prim = result.first;
 
+    auto pd = (*prim).get_primitive_desc();
+    auto query_md = [&pd](dnnl::query what, int idx = 0) -> dnnl::memory::desc {
+        auto query = dnnl::convert_to_c(what);
+        const dnnl_memory_desc_t *cdesc = dnnl_primitive_desc_query_md(pd, query, idx);
+        if (!cdesc)
+            IE_THROW() << "query_md failed for query=" << query << " idx=" << idx << ".";
+        return dnnl::memory::desc(*cdesc);
+    };
```

Comment:
M.b. it does make sense to move this code to the ScratchPad class as a static utility function?

Reply:
Also such function might return `DnnlMemoryDescPtr` and `setScratchPad` might take this type instead of oneDNN native. Thus we isolate oneDNN stuff and use OV entities.

Reply:
changed